### PR TITLE
[#6251,#6256,#7398,#7506,#7507,#7509,#7510,#7511,#7512] some CMake TLC (main)

### PR DESCRIPTION
**Pull request by SwooshyCueb**
In service of #6251
In service of #6256
Addresses #7398
Addresses #7506
Addresses #7507
Addresses #7509
Addresses #7510
Addresses #7511
Addresses #7512

I started preparing for spdlog's transition from static library to shared library, and figured I this was a good opportunity to move it over to `find_package` as well. That meant also moving fmt to `find_package`, and in the process of doing so I found a bunch of other things I could fix at the same time.

Not tested yet.

### CODE REVIEW

**Comment by korydraughn** [comment]
Excellent. Happy to see things move towards `find_package`.

File: lib/core/CMakeLists.txt [hunk 1]
```diff
@@ -79,6 +79,7 @@ target_link_libraries(
   PUBLIC
   nlohmann_json::nlohmann_json
   fmt::fmt
+  spdlog::spdlog_header_only # TODO: switch to spdlog::spdlog once shared lib is in externals
```
Comment (alanking) [inline lib/core/CMakeLists.txt:82]: RE TODO: We have an issue for this already, I assume?

**Review by alanking — COMMENTED** [review]
Nice. Looking more official by the day.

Should this commit message...
"[7508] CMake: do not call target_include_directories on irods_server"
...be linked to #7509 instead?

Also, the commit "[6256] CMake: use find_package for spdlog" has a few things that look like they belong to other commits, if that matters. e.g. Some more `CMAKE_SOURCE_DIR` and `fmt::fmt` cleanup efforts.

File: lib/core/CMakeLists.txt (hunk 1 above)
Comment (SwooshyCueb) [reply lib/core/CMakeLists.txt:82]: No, I'm just going to keep #6256 open

**Comment by SwooshyCueb** [comment]
I'm going to have to re-think my approach to #7398 and #7505, as both would necessitate downstream projects calling `find_package(IRODS` *after* `project`. This is troublesome, as our CMake module that sets the compiler needs to be included *before* the `project` call.

**Comment by SwooshyCueb** [comment]
> Should this commit message... "[7508] CMake: do not call target_include_directories on irods_server" ...be linked to #7509 instead?

Yup, nice catch.

> Also, the commit "[6256] CMake: use find_package for spdlog" has a few things that look like they belong to other commits, if that matters. e.g. Some more `CMAKE_SOURCE_DIR` and `fmt::fmt` cleanup efforts.

I'll clean that up too, nice catch.

**Comment by SwooshyCueb** [comment]
I'm dropping #7505 from this PR, as we can't define targets in our config file until we have moved away from setting the compilers in our cmake files.

File: lib/core/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (SwooshyCueb) [reply lib/core/CMakeLists.txt]: Covered by #6256

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by SwooshyCueb** [comment]
\#'d

**Review by alanking — APPROVED** [review]

**Pull request merged**